components/browser/state/src/main/java/mozilla/components/browser/state/action/BrowserAction.kt

```diff
@@ -574,12 +574,12 @@ sealed class DownloadAction : BrowserAction() {
     /**
      * Updates the [BrowserState] to track the provided [download] as added.
      */
-    data class AddDownloadAction(val download: DownloadState) : DownloadAction()
+    data class AddDownloadAction(val download: DownloadState, val restored: Boolean = false) : DownloadAction()
```

Comment:
Could `restored` just be part of the `DownloadState` instead of being in the action?

Reply:
Do you think that we are going to need it in the future?

Reply:
This is the only item left on the updated. I'm not sure if we are going to need `restored`  on `DownloadState` in the future, as now it just serves the purpose of identifying a download is coming from the storage.

Reply:
👍

Reply:
Added a new `RestoreDownloadAction` makes things clear :)